causal_automl/fetch_eia_metadata.py

```diff
@@ -0,0 +1,179 @@
+#!/usr/bin/env python3
+"""
+Fetch metadata from the EIA v2 API and upload it to S3.
+
+Usage:
+> python fetch_eia_metadata.py --category electricity --api_key <API_KEY> --aws_profile
+
+This script traverses the EIA v2 API under a specified category and collects all time series
+metadata. It then writes the extracted metadata to an S3 bucket in CSV format.
+
+Arguments:
+    --category       Root category path under the EIA v2 API.
+    --api_key        EIA API key used to authenticate requests.
+    --aws_profile    AWS CLI profile to use for writing the output to S3. Defaults to "ck".
```

Comment:
DRY suggests to merge this with the rest of the command line options. No need to replicate the info.